src/flash.rs

```diff
@@ -20,6 +303,37 @@ impl FlashExt for FLASH {
 pub struct Parts {
     /// Opaque ACR register
     pub acr: ACR,
+
+    /// Opaque AR register
+    pub(crate) ar: AR,
+
+    /// Opaque CR register
+    pub(crate) cr: CR,
+
+    /// Opaque KEYR register
+    pub(crate) keyr: KEYR,
+
+    /// Opaque OBR register
+    pub(crate) obr: OBR,
+
+    /// Opaque OPTKEYR register
+    pub(crate) optkeyr: OPTKEYR,
+
+    /// Opaque SR register
+    pub(crate) sr: SR,
+
+    /// Opaque WRPR register
+    pub(crate) wrpr: WRPR,
+}
+impl Parts {
+    pub fn writer(&mut self, sector_sz: SectorSize, flash_sz: FlashSize) -> FlashWriter {
```

Comment:
Unless I'm missing something, `&mut self` only lasts until the function returns, but the `FlashWriter` lasts much longer than that.

A `release` method sounds like the best option to me. We could probably also expose the functionality you talk about through the flash writer if required, right?

Reply:
I just tested, it won't let us call writer() again. This is because the FlashWriter struct contains a mutable reference to Parts.

Reply:
We definitely could expose the extra functionality through the FlashWriter if we really needed to, but I think it would make sense to have it separate. In any case, I think we can rely of reference lifetimes to ensure safe/correct usage around the FlashWriter, Parts, and the writer() function.